pkg/virt-launcher/virtwrap/api/defaults.go

```diff
@@ -25,6 +25,8 @@ func (d *Defaulter) SetDefaults_OSType(ostype *OSType) {
 	if ostype.Arch == "" {
 		if d.Architecture == "ppc64le" {
 			ostype.Arch = "ppc64le"
+		} else if d.Architecture == "arm64" || d.Architecture == "aarch64" {
```

Comment:
Here we have the right approach regarding to the architecture and testability.

Reply:
Ok, I get it.

Reply:
> Here we have the right approach regarding to the architecture and testability.

As CPU architecture setting is used by virt-launcher, virt-config and virt-controller, shall we write it in pkg/util?

Reply:
I see different possibilities:

 * If a lot of differences occur between architectures, it would make sense to have different implementations against interfaces (I don't see this applicable, they are almost 100% identical)
  * Use structs or function parameters where you pass the architecture: Just pass on the edge of the application `runtime.GOARCH` to structs or functions to propagate it  through the code.
  
I think the second case applies here. But maybe I don't understand what you mean.

Reply:
Ok, I get your idea